team_picker/routes/cohorts.js

```diff
@@ -0,0 +1,109 @@
+const express = require("express");
+const knex = require("../db/client"); // This allows us to interact with the database
+const router = express.Router();
+const pickerLogic = require("../public/pickerLogic.js");
+
+
+
+  // render cohort index
+  router.get("/index", (req, res) => {
+    knex("cohorts")
+      .orderBy("created_at", "desc")
+      .then(cohorts => {
+        res.render("cohorts/index", { cohorts: cohorts });
+      });
+  });
+
+
+//Render new cohort template
+router.get("/", (req, res) => {
+  res.render("cohorts/new", {cohort: false});
+});
+
+// create new cohort
+router.post("/", (req, res) => {
+    const formData = req.body;
+    knex("cohorts")
+      .insert({
+        name: formData.name,
+        members: formData.members,
+        logourl: formData.logourl
+      })
+      .returning("*")
+      .then(cohorts => {
+        const cohort = cohorts[0];
+        res.redirect(`/cohorts/${cohort.id}`);
+      });
+  });
+
+
+
+//show single cohort
+router.get("/:id", (req, res) => {
+    const id = req.params.id;
+    knex("cohorts")
+      .where("id", id)
+      .first()
+      .then(cohort => {
+        if (!cohort) {
+          res.send("No Cohort found")
+        } else {
+          if (req.query.teams) {
```

Comment:
It seems you were too focused on getting the logic implemented to assign teams. This will not work when the flows are:
* redirect to show page from new on submit
* clicking on a post from index page to view
because these two scenarios will never have the query unless you explicitly append to line 35 (which seems weird already). 

One way you can get around this is by adding an else block to line 61 and it will have something like `res.render("cohorts/show", { cohort, finalTeams: [] })`